### PR TITLE
:sparkles: Data pages: enable rendering of data pages on all eligible data pages, remove old datapage

### PR DESCRIPTION
**Pull request by danyx23**
This PR removes the old prototype data page code and switches rendering of eligible charts to show data pages.

☣️ should only be merged once the team is happy with this as it will start rendering data pages on public /grapher/ urls

### CODE REVIEW

**Comment by danyx23** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2848** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2848" target="_blank">Graphite</a>  👈
    * **PR #2739** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2739" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/2848?utm_source=stack-comment).

File: baker/GrapherBaker.tsx [hunk 1]
```diff
@@ -82,123 +75,32 @@ export const renderDataPageOrGrapherPage = async (
...
-        logErrorAndMaybeSendToBugsnag(
-            new JsonError(
-                `${dataPageNotPublishedError} Please check that the googleDocEditLink ${datapageJson.googleDocEditLink} is correct.`
-            )
-        )
-        return renderGrapherPage(grapher)
-    }
+    // If we have a single Y variable and that one has a schema version >= 2,
+    // meaning it has the metadata to render a datapage, then we show the datapage
+    // based on this information. Otherwise we see if there is a legacy datapage.json
```
Comment (sophiamersmann) [inline baker/GrapherBaker.tsx:80]: Support for legacy datapage.json configs is dropped, no?

Would also be good to update the code comment above 👆

File: baker/GrapherBaker.tsx [hunk 2]
```diff
@@ -82,123 +75,32 @@ export const renderDataPageOrGrapherPage = async (
...
-        )
-        return renderGrapherPage(grapher)
+        if (
+            variableMetadata.schemaVersion !== undefined &&
+            variableMetadata.schemaVersion >= 2
+        ) {
+            return await renderDataPageV2({
+                variableId,
+                variableMetadata,
+                isPreviewing: true,
```
Comment (sophiamersmann) [inline baker/GrapherBaker.tsx:96]: Is this right?

**Review by sophiamersmann — APPROVED** [review]
Looks good to me!

If it's easy to do, should we rename DataPageV2 to DataPage?

File: baker/GrapherBaker.tsx (hunk 2 above)
Comment (danyx23) [reply baker/GrapherBaker.tsx:96]: No the code as is works as intended - this is in the function that shows the preview so we do want to set isPreviewing to true here

**Comment by danyx23** [comment]
### Merge activity

* **Nov 27, 6:24 AM**: @danyx23 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/2848).

**Pull request merged**